### PR TITLE
Improve Rate Limits' Documentation

### PR DESCRIPTION
**Pull request by mirsaeedi**
I added an example and a bit of explanation about `GitHubClient.Miscellaneous.GetRateLimits()`.  #1737

### CODE REVIEW

File: docs/getting-started.md [hunk 1]
```diff
@@ -90,6 +92,23 @@ var howManyRequestsDoIHaveLeft = rateLimit?.Remaining;
...
 
+However, if in some cases you need to get the Rate Limit directly from Github, you should call `GitHubClient.Miscellaneous.GetRateLimits()`.
+
+Example usage:
+
+```csharp
+GithubClient client; 
+//Create & initialize the client here
+
+var miscellaneousRateLimit = client.Miscellaneous.GetRateLimits();
```
Comment (ryangribble) [inline docs/getting-started.md]: Can we use,`await` syntax here to avoid the `.Result` below?

File: docs/getting-started.md [hunk 2]
```diff
@@ -90,6 +92,23 @@ var howManyRequestsDoIHaveLeft = rateLimit?.Remaining;
...
+
+var rate = miscellaneousRateLimit.Result.Rate;
```
Comment (ryangribble) [inline docs/getting-started.md]: According to the docs the `Rate` property is deprecated so our docs should actually steer people towards the proper field (`.Resource.Core`)

Comment (ryangribble) [reply docs/getting-started.md]: Could probably also mention the search API rate limit in the code sample too

**Comment by ryangribble** [comment]
Closes #1737

**Comment by mirsaeedi** [comment]
@ryangribble  I tried to satisfy your points in my new commit.

File: docs/getting-started.md (hunk 2 above)
Comment (mirsaeedi) [reply docs/getting-started.md]: I replaced the `Rate` property with the `.Resource.Core`. Also, I added a sample for the search API.

**Comment by ryangribble** [comment]
Thanks so much @mirsaeedi for tidying up this doco!

LGTM

**Comment by ryangribble** [comment]
release_notes: Updated Rate Limits documentation and samples

**Pull request merged**